src/task/task.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package taskapi;
+option go_package = "github.com/pachyderm/pachyderm/v2/src/task";
+
+import "gogoproto/gogo.proto";
+
+enum State {
+  UNKNOWN = 0;
+  RUNNING = 1;
+  SUCCESS = 2;
+  FAILURE = 3;
+  CLAIMED = 4; // not a real state used by task logic
+}
+
+message TaskInfo {
+  string id = 1 [(gogoproto.customname) = "ID"];
+  string full_key = 3;
```

Comment:
The concern I had with namespace and group is how to extract it properly from the key. If you can assume the group has no slashes, we can just split the string. Otherwise it seems like we in general need to look for the group collection by trying different prefixes. Is there some simple thing I'm missing here?